contracts/ConnextModule.sol

```diff
@@ -50,47 +53,56 @@ contract ConnextModule is Module {
             address _target,
             address _originAddress,
             uint32 _origin,
-            address _executor
+            address _connext
         ) = abi.decode(initializeParams, (address, address, address, address, uint32, address));
 
         setAvatar(_avatar);
         setTarget(_target);
         setOriginAddress(_originAddress);
         setOrigin(_origin);
-        setExecutor(_executor);
+        setConnext(_connext);
         transferOwnership(_owner);
 
-        emit ModuleSetUp(owner(), avatar, target, originAddress, origin, executor);
+        emit ModuleSetUp(owner(), avatar, target, originAddress, origin, connext);
     }
 
-    modifier onlyExecutor(bytes memory _message) {
-        if (msg.sender != executor) revert ExecutorOnly();
-        if (LibCrossDomainProperty.originSender(_message) != originAddress) revert OriginAddressOnly();
-        if (LibCrossDomainProperty.origin(_message) != origin) revert OriginOnly();
+    modifier onlyConnext(address _originSender, uint32 _origin) {
+        if (msg.sender != connext) revert ConnextOnly();
+        if (_originSender != originAddress) revert OriginAddressOnly();
+        if (_origin != origin) revert OriginOnly();
         _;
     }
 
-    function execute(bytes calldata _message) external onlyExecutor(_message) {
-        uint256 length = LibCrossDomainProperty.callDataLength(bytes29(_message));
+    function xReceive(
+        bytes32 _transferId,
+        uint256 _amount,
+        address _asset,
+        address _originSender,
+        uint32 _origin,
+        bytes memory _callData
+    ) 
+        external onlyConnext(_originSender, _origin)
+        returns (bytes memory) {
+            _execute(_callData);
+        
+    }
+
+    function _execute(bytes calldata _message) internal {
         (address _to, uint256 _value, bytes memory _data, Enum.Operation _operation) = abi.decode(
-            bytes(_message[0:length]),
+            _message,
             (address, uint256, bytes, Enum.Operation)
         );
         if (!exec(_to, _value, _data, _operation)) revert ModuleTransactionFailed();
     }
 
     function setOriginAddress(address _originAddress) public onlyOwner {
+        require(_originAddress != address(0), "Sender should not be address(0)");
         originAddress = _originAddress;
         emit OriginAddressSet(originAddress);
     }
 
-    function setOrigin(uint32 _origin) public onlyOwner {
-        origin = _origin;
-        emit OriginSet(origin);
-    }
```

Comment:
do we want to include this still? if not,  we should delete the `OriginSet` event

Reply:
meant to include this still!